binders/kafka-binder/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -1516,6 +1524,19 @@ public void addProducerConfigCustomizer(ProducerConfigCustomizer producerConfigC
 				setSendTimeoutExpression(producerProperties.getExtension().getSendTimeoutExpression());
 			}
 			this.producerFactory = producerFactory;
+
+			/*
+			 	Activate kafkaPartitionHandler if 'getPartitionKeyExpression' was used to recalculate/override
+			 	partition with current partition size from kafkaTemplate just before message is send.
+			 	PartitionKeyExpression 'payload' is not supported here,
+			 	because of OutboundContentTypeConvertingInterceptor was called before.
```

Comment:
This comment needs to be clarified. Could you tell me why you are guarding the `payload` key expression? What will happen if we specify `payload` as the `partition-key-expression`? Not sure how the `OutboundContentTypeConvertingInterceptor` is related here.

Reply:
I had to add a KafkaPartitionHandler within handleMessage because of architectual reasons (see my next/other reply). The disadvantage of this is that the payload of a message is already encoded by OutboundContentTypeConvertingInterceptor before and can't be read anymore here. A producer configuration with payload as PartitionKeyExpression can't be handled by my fix therefore.